app/controllers/submissions_controller.rb

```diff
@@ -809,4 +839,20 @@ def get_all_file_data(revision, grouping, path)
   def parent_params
     params[:grouping_id].nil? ? super : [*super, :grouping_id]
   end
+
+  # Helper that extracts the URL from a url file to send to a user.
+  # If no URL is found, returns a string saying the file cannot be displayed.
+  def extract_url(file_content)
```

Comment:
Right, so I see what you're trying to do here but I again think it's a bit more complicated than necessary. Maybe it's a holdover from earlier when the file format was more complex, but now the file contains just the URL string. Replacing an invalid URL with an error message isn't very helpful, especially when the current error message says "Please download it to view", which isn't applicable to URLs.

Instead, please remove this function and just send the contents of the file when it's requested, and leave the front-end and user to see and handle any invalid URLs.

(You can still keep the `is_valid_url?` for when saving the URL file. That's good!)